modules/Bio/EnsEMBL/RNAProduct.pm

```diff
@@ -0,0 +1,976 @@
+                   "CREATED_DATE", "MODIFIED_DATE"], @_);
+
+  # For consistency between stable_id() and stable_id_version()
+  $stable_id //= '';
+
+  # Default version
+  $version //= 1;
+
+  my $self = bless {
+    'start'      => $seq_start,
+    'end'        => $seq_end,
+    'start_exon' => $start_exon,
+    'end_exon'   => $end_exon,
+    'stable_id'  => $stable_id,
+    'version'    => $version,
+    'dbID'       => $dbID,
+    'seq'        => $seq,
+    'created_date' => $created_date,
+    'modified_date' => $modified_date,
+    'type_code'  => $type_code,
+  }, $class;
+
+  $self->adaptor($adaptor);
+
+  return $self;
+}
+
+
+=head2 add_Attributes
+
+  Arg [1..N] : Bio::EnsEMBL::Attribute $attribute
+               Attributes to add.
+  Example    : $rnaproduct->add_Attributes($selenocysteine_attribute);
+  Description: Adds an Attribute to the RNAProduct.
+               If you add an attribute before you retrieve any from database,
+               lazy load will be disabled.
+  Returntype : none
+  Exceptions : throw on incorrect arguments
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub add_Attributes {
+  my ($self, @attribs) = @_;
+
+  if (! exists $self->{'attributes'}) {
+    $self->{'attributes'} = [];
+  }
+
+  for my $attrib (@attribs) {
+    if (! $attrib->isa("Bio::EnsEMBL::Attribute")) {
+      throw("Argument to add_Attribute must be a Bio::EnsEMBL::Attribute");
+    }
+    push (@{$self->{'attributes'}}, $attrib);
+    # FIXME: why is this here???
+    $self->{seq}=undef;
+  }
+
+  return;
+}
+
+
+=head2 add_DBEntry
+
+  Arg [1]    : Bio::EnsEMBL::DBEntry $dbe
+               The dbEntry to be added
+  Example    : $rnaproduct->add_DBEntry($xref);
+  Description: Associates a DBEntry with this rnaproduct. Note that adding
+               DBEntries will prevent future lazy-loading of DBEntries for this
+               rnaproduct (see get_all_DBEntries).
+  Returntype : none
+  Exceptions : thrown on incorrect argument type
+  Caller     : general
+  Status     : Stable
+
+=cut
+
+sub add_DBEntry {
+  my ($self, $dbe) = @_;
+
+  if (!$dbe || !ref($dbe) || !$dbe->isa('Bio::EnsEMBL::DBEntry')) {
+    throw('Expected DBEntry argument');
+  }
+
+  $self->{'dbentries'} ||= [];
+  push @{$self->{'dbentries'}}, $dbe;
+
+  return;
+}
+
+
+=head2 cdna_end
+
+    Example     : $rnaproduct_cdna_end = $rnaproduct->cdna_end();
+    Description : Returns the end position of the rnaproduct in cDNA
+                  coordinates.
+                  Since rnaproducts do not span multiple exons, this is
+                  simply an alias for end().
+    Return type : Integer
+    Caller      : General
+    Status      : Stable
+
+=cut
+
+sub cdna_end {
+  my $self = shift;
+
+  return $self->end();
+}
+
+
+=head2 cdna_start
+
+    Example     : $rnaproduct_cdna_start = $rnaproduct->cdna_start();
+    Description : Returns the start position of the rnaproduct in cDNA
```

Comment:
rnaproduct => RNA product
or should it be $rnaproduct or RNAProduct as used in previous pods?